server/service/integration_core_test.go

```diff
@@ -7900,3 +7900,273 @@ func (s *integrationTestSuite) TestHostsReportWithPolicyResults() {
+	require.NoError(t, gqrr.Err)
+	require.Equal(t, usbDevicesQuery.ID, gqrr.QueryID)
+	require.Len(t, gqrr.Results, 2)
+	require.Equal(t, host2Team1.ID, gqrr.Results[0].HostID)
+	require.Equal(t, host2Team1.Hostname, gqrr.Results[0].Hostname)
+	require.NotZero(t, gqrr.Results[0].LastFetched)
+	require.Equal(t, map[string]string{
+		"class":       "239",
+		"model":       "HD Pro Webcam C920",
+		"model_id":    "0892",
+		"protocol":    "",
+		"removable":   "1",
+		"serial":      "zoobar",
+		"subclass":    "2",
+		"usb_address": "3",
+		"usb_port":    "1",
+		"vendor":      "",
+		"vendor_id":   "046d",
+		"version":     "0.19",
+	}, gqrr.Results[0].Columns)
+	require.Equal(t, host2Team1.ID, gqrr.Results[1].HostID)
+	require.Equal(t, host2Team1.Hostname, gqrr.Results[1].Hostname)
+	require.NotZero(t, gqrr.Results[1].LastFetched)
+	require.Equal(t, map[string]string{
+		"class":       "0",
+		"model":       "Apple Internal Keyboard / Trackpad",
+		"model_id":    "027e",
+		"protocol":    "",
+		"removable":   "0",
+		"serial":      "foobar",
+		"subclass":    "0",
+		"usb_address": "8",
+		"usb_port":    "5",
+		"vendor":      "Apple Inc.",
+		"vendor_id":   "05ac",
+		"version":     "9.33",
+	}, gqrr.Results[1].Columns)
+
+	gqrr = getQueryReportResponse{}
+	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/queries/%d/report", osqueryInfoQuery.ID), getQueryReportRequest{}, http.StatusOK, &gqrr)
+	require.NoError(t, gqrr.Err)
+	require.Equal(t, osqueryInfoQuery.ID, gqrr.QueryID)
+	require.Len(t, gqrr.Results, 2)
+	require.Equal(t, host1Global.ID, gqrr.Results[0].HostID)
+	require.Equal(t, host1Global.Hostname, gqrr.Results[0].Hostname)
+	require.NotZero(t, gqrr.Results[0].LastFetched)
+	require.Equal(t, map[string]string{
+		"build_distro":   "centos7",
+		"build_platform": "linux",
+		"config_hash":    "eed0d8296e5f90b790a23814a9db7a127b13498d",
+		"config_valid":   "1",
+		"extensions":     "active",
+		"instance_id":    "e5799132-85ab-4cfa-89f3-03e0dd3c509a",
+		"pid":            "3574",
+		"platform_mask":  "9",
+		"start_time":     "1696502961",
+		"uuid":           host1Global.UUID,
+		"version":        "5.9.1",
+		"watcher":        "3570",
+	}, gqrr.Results[0].Columns)
+	require.Equal(t, host2Team1.ID, gqrr.Results[1].HostID)
+	require.Equal(t, host2Team1.Hostname, gqrr.Results[1].Hostname)
+	require.NotZero(t, gqrr.Results[1].LastFetched)
+	require.Equal(t, map[string]string{
+		"build_distro":   "10.14",
+		"build_platform": "darwin",
+		"config_hash":    "eed0d8296e5f90b790a23814a9db7a127b13498d",
+		"config_valid":   "1",
+		"extensions":     "active",
+		"instance_id":    "7f02ff0f-f8a7-4ba9-a1d2-66836b154f4a",
+		"pid":            "95637",
+		"platform_mask":  "21",
+		"start_time":     "1696611201",
+		"uuid":           host2Team1.UUID,
+		"version":        "5.9.1",
+		"watcher":        "95636",
+	}, gqrr.Results[1].Columns)
+}
```

Comment:
nice!  there's unit tests for things like handling multiple responses for the same queryID, and different scenarios to determine when to discard results.  should we add those to the integration suite as well?